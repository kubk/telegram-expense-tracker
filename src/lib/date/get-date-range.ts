import { DateTime } from 'luxon';

export const getDateRangeFromWeek = (year: number, weekNumber: number) => {
  const dateTime = DateTime.fromObject({
    weekYear: year,
    weekNumber: weekNumber,
  });

  if (!dateTime.isValid) {
    throw new Error(
      `Unable to parse date from week ${weekNumber}, year ${year}`
    );
  }

  return {
    from: dateTime.startOf('week').toJSDate(),
    to: dateTime.endOf('week').toJSDate(),
  };
};

export const getDateRangeFromMonth = (year: number, month: number) => {
  const dateTime = DateTime.fromObject({
    month: month,
    year: year,
  });

  if (!dateTime.isValid) {
    throw new Error(`Unable to parse date from month ${month}, year ${year}`);
  }

  return {
    from: dateTime.startOf('month').toJSDate(),
    to: dateTime.endOf('month').toJSDate(),
  };
};
