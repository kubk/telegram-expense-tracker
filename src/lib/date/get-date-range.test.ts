import { getDateRangeFromMonth } from './get-date-range';

test('get date range from month', () => {
  expect(getDateRangeFromMonth(2022, 1)).toMatchSnapshot();

  expect(() => getDateRangeFromMonth(2022, 99)).toThrow();
});
