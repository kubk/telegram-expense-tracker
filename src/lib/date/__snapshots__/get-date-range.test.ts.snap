// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get date range from month 1`] = `
Object {
  "from": 2022-01-01T00:00:00.000Z,
  "to": 2022-01-31T23:59:59.999Z,
}
`;

exports[`get date range from week 1`] = `
Object {
  "from": 2022-01-03T00:00:00.000Z,
  "to": 2022-01-09T23:59:59.999Z,
}
`;

exports[`get date range from week 2`] = `
Object {
  "from": 2022-01-10T00:00:00.000Z,
  "to": 2022-01-16T23:59:59.999Z,
}
`;

exports[`get date range from week 3`] = `
Object {
  "from": 2022-01-17T00:00:00.000Z,
  "to": 2022-01-23T23:59:59.999Z,
}
`;
